apps/system/lockscreen/js/lockscreen_state_manager.js

```diff
@@ -453,6 +487,11 @@
         // @see: SecureAppManager
         this.onSecureAppClosing();
         break;
+      case 'keyup':
+        if(this.lockScreenStates.passcodeEnabled && (evt.keyCode >= KEY_CODE_ZERO && evt.keyCode <= KEY_CODE_NINE)) {
```

Comment:
As soon as decision most probably won't be made regarding this case, you shall leave ability to dismiss lockscreen (when 'pin lock' disabled) for example by pressing on any hardware key. Otherwise your feature will make device/emulator really hard to use (at least for mentioned configuration: {lock: ON; pin: OFF}).